src/libraries/Microsoft.Extensions.Hosting.Systemd/src/SystemdHelpers.cs

```diff
@@ -20,16 +20,25 @@ public static partial class SystemdHelpers
         /// </summary>
         /// <returns><c>True</c> if the current process is hosted as a systemd Service, otherwise <c>false</c>.</returns>
         public static bool IsSystemdService()
-            => _isSystemdService ?? (bool)(_isSystemdService = CheckParentIsSystemd());
+            => _isSystemdService ?? (bool)(_isSystemdService = GetIsSystemdService());
```

Comment:
While we are here changing this line, this can be

```suggestion
            => _isSystemdService ??= GetIsSystemdService();
```